python/interpret_community/widget/explanation_dashboard.py

```diff
@@ -0,0 +1,183 @@
+from gevent.pywsgi import WSGIServer
+from flask import Flask, render_template, request
+from IPython.display import display, IFrame
+import threading
+import socket
+import requests
+import os
+import json
+from .explanation_dashboard_input import ExplanationDashboardInput
+
+"""Explanation Dashboard Class.
+
+:param explanation: An object that represents an explanation.
+:type explanation: ExplanationMixin
+:param model: An object that represents a model. It is assumed that for the classification case
+    it has a method of predict_proba() returning the prediction probabilities for each
+    class and for the regression case a method of predict() returning the prediction value.
+:type model: object
+:param dataset:  A matrix of feature vector examples (# examples x # features), the same samples
+    used to build the explanation. Will overwrite any set on explanation object already
+:type dataset: numpy.array or list[][]
+:param true_y: The true labels for the provided dataset. Will overwrite any set on
+    explanation object already
+:type true_y: numpy.array or list[]
+:param classes: The class names
+:type classes: numpy.array or list[]
+:param features: Feature names
+:type features: numpy.array or list[]
+:param port: the port to use on localhosted service
+:type port: number
+:param use_cdn: should load latest dashboard script from cdn, fall back to local script if false
+:type use_cdn: boolean
+"""
+
+
+class ExplanationDashboard:
+    service = None
+    explanations = {}
+    model_count = 0
+    _cdn_path = "v0.1.js"
+
+    class DashboardService:
```

Comment:
this class is quite large, maybe move it to a separate file?  it's confusing to read nested classes.  You can always add an _ if you don't want it to be public

Reply:
I see, yes, the goal is that this is a private class that the main class in this file can create to have a singleton of. I can move it, but the main goal is to have this file have access to it and not expose it.

Reply:
I'm not sure how to resolve what looks to be some circular dependencies in removing the class. Is it ok to leave this in place for now, and extract later?